docs/guided-remediation.md

```diff
@@ -0,0 +1,225 @@
+UPGRADED-PACKAGE: minimist,1.2.0,1.2.8
+UPGRADED-PACKAGE: acorn,5.7.3,5.4.1
+UPGRADED-PACKAGE: acorn,6.0.2,6.4.2
+UPGRADED-PACKAGE: acorn,7.1.0,7.4.1
+UPGRADED-PACKAGE: ansi-regex,3.0.0,3.0.1
+UPGRADED-PACKAGE: dat.gui,0.7.3,0.7.9
+UPGRADED-PACKAGE: get-func-name,2.0.0,2.0.2
+UPGRADED-PACKAGE: glob-parent,5.0.0,5.1.2
+UPGRADED-PACKAGE: json5,2.1.0,2.2.3
+UPGRADED-PACKAGE: minimist,1.2.5,1.2.8
+UPGRADED-PACKAGE: node-fetch,2.6.6,2.7.0
+UPGRADED-PACKAGE: path-parse,1.0.6,1.0.7
+UPGRADED-PACKAGE: pathval,1.1.0,1.1.1
+UPGRADED-PACKAGE: qs,6.5.2,6.5.3
+UPGRADED-PACKAGE: semver,5.5.1,5.7.2
+UPGRADED-PACKAGE: semver,5.6.0,5.7.2
+UPGRADED-PACKAGE: semver,6.3.0,6.3.1
+UPGRADED-PACKAGE: ws,6.2.1,6.2.2
+UPGRADED-PACKAGE: ws,7.1.2,7.5.9
+UPGRADED-PACKAGE: y18n,4.0.0,4.0.3
+REMAINING-VULNS: 29
+UNFIXABLE-VULNS: 29
+Rewriting path/to/package-lock.json...
+```
+</details>
+
+<details markdown="1">
+<summary><b>Relock Strategy</b></summary>
+```bash
+osv-scanner fix --non-interactive --strategy=relock --apply-top=2 -M path/to/package.json -L path/to/package-lock.json
+```
+```
+Resolving path/to/package.json...
+Found 11 vulnerabilities matching the filter
+Can fix 4/11 matching vulnerabilities by changing 2 dependencies
+UPGRADED-PACKAGE: mocha,^5.2.0,^9.2.2
+UPGRADED-PACKAGE: @google-cloud/cloudbuild,^2.6.0,^3.10.0
+REMAINING-VULNS: 7
+UNFIXABLE-VULNS: 3
+Rewriting path/to/package.json...
+Shelling out to regenerate lockfile...
+Executing `npm install --package-lock-only`...
+[ npm output ]
+```
+</details>
+
+## Remediation Flags
+
+The `fix` subcommand has a number of flags to allow you to control which vulnerabilities and patches may be considered during remediation.
+
+### Vulnerability Selection
+
+The following flags may be used to filter which vulnerabilities will be selected for remediation:
+
+- `--max-depth=<value>`: The maximum dependency depth to consider vulnerabilities from.
+
+  For example, `--max-depth=1` will only consider vulnerabilities affecting direct dependencies, while `--max-depth=2` will consider up to the direct dependencies of your direct dependencies. Setting `--max-depth=-1` will consider all dependencies, regardless of depth.
+
+- `--min-severity=<value>`: The minimum CVSS score of vulnerabilities to consider.
+
+  For example, `--min-severity=7.5` will consider only vulnerabilities with CVSS a score of 7.5 and above. If the OSV record for the vulnerability does not have a CVSS score, it will **not** be excluded.
+
+- `--ignore-dev`: Whether to exclude vulnerabilities from dependencies that are only used in `devDependencies`.
+- `--ignore-vulns=<comma-separated list of IDs>`: A list of OSV IDs to exclude from consideration.
+- `--vulns=<comma-separated list of IDs>`: An explicit list of OSV IDs that should exclusively be considered, as long as the other criteria are satisfied.
+
+A vulnerability is only considered if it satisfies all the conditions set by these flags.
+
+### Dependency Upgrade Options
```

Comment:
do we still consider to refactor this part of flags to only one flag?

Reply:
I would like to, but I need to implement it first :slightly_smiling_face: